Process queue on _afterEnd event, not 'end'

isaacs opened the pull request:
This should fix https://github.com/isaacs/node-tap/issues/153

@sam-github would you be so kind as to review and make sure that it addresses your problem?

Thanks!

isaacs committed:
test showing incorrect timing behavior of end event

isaacs committed:
Process the queue on private _afterEnd event, not end

This way, a user can listen to 'end' and their handler will be called
before moving on to the next test in the queue.

Fix #153

rmg commented:
Is there any concern that relying on the event handler being synchronous turns this into more of a hook point than a notification? Not sure if that's a distinction anyone actually cares about..

Does seem to address @sam-github's issue, though, as far as I can tell.

isaacs commented:
Yeah, actually, that shouldn't even be an "event", just a method or something.

isaacs committed:
Replace _afterEnd event with method call

It's internal and only used in one specific way.  No need to be
using the EventEmitter machinery for this.
